### PR TITLE
Document `eos.ObservableCache` and use it to speed up the `predict_observables` task

### PR DESCRIPTION
**Pull request by dvandyk**

### CODE REVIEW

File: python/_eos.cc [hunk 1]
```diff
@@ -498,11 +498,28 @@ BOOST_PYTHON_MODULE(_eos)
...
-        .def("add", &ObservableCache::add)
-        .def("update", &ObservableCache::update)
+        .def("__getitem__", &ObservableCache::operator[], R"(
+            Access the cached value of an observable.
+
+            :param handle: The handle of the observable.
+            :type handle: int
+        )", args("handle"))
+        .def("add", &ObservableCache::add, R"(
+            Adds an existing observable to the cache.
```
Comment (mreboud) [inline python/_eos.cc:512]: For consistency I would either remove the `s` here or add it in the previous definition.

File: python/_eos.cc [hunk 2]
```diff
@@ -498,11 +498,28 @@ BOOST_PYTHON_MODULE(_eos)
...
+
+            :param observable: The observable to add to the cache.
+            :type observable: eos.Observable
+
+            :returns: An internal handle to the cached observable. The observable's value can be retrieved using ``cache[handle]``.
+            :rtype: int
+        )", args("observable"))
+        .def("update", &ObservableCache::update, R"(
+            Updates the cache for the current parameter point.
```
Comment (mreboud) [inline python/_eos.cc:521]: Ditto

**Review by mreboud — APPROVED** [review]
Approved (with the exception of the typo) :)

**Comment by dvandyk** [comment]
> Approved (with the exception of the typo) :)

Thanks. Typos in the documentation fixed.

**Pull request merged**